src/parsing.jl

```diff
@@ -683,12 +683,13 @@ end
 
 struct FallbackConversion <: AbstractDict{Tuple{Oid,Type},Base.Callable} end
 
+struct ParseType{T} <: Function end
```

Comment:
Have you seen this subtyping before in other packages? It seems like it could be dangerous.

Reply:
I have not. Any idea what might go wrong?

Reply:
Some functions that dispatch on `Function` (like `Base.show_function`) use the internal structure of the function type. However, that particular example works since the internal structure of a struct type and a function type are similar. 

In any case, I found an example using this `<: Function` in the wild, so I'm comfortable with it: https://github.com/invenia/Impute.jl/blob/9230661ae5f3dc828fea58b32970c874574cb654/src/chain.jl#L13